src/Authorization.AspNetCore/AuthorizationValidationRule.cs

```diff
@@ -103,100 +104,21 @@ private async Task AuthorizeAsync(INode node, IProvideMetadata type, ValidationC
-            {
-                case ClaimsAuthorizationRequirement claimsAuthorizationRequirement:
-                    error.Append("Required claim '");
-                    error.Append(claimsAuthorizationRequirement.ClaimType);
-                    if (claimsAuthorizationRequirement.AllowedValues == null || !claimsAuthorizationRequirement.AllowedValues.Any())
-                    {
-                        error.Append("' is not present.");
-                    }
-                    else
-                    {
-                        error.Append("' with any value of '");
-                        error.Append(string.Join(", ", claimsAuthorizationRequirement.AllowedValues));
-                        error.Append("' is not present.");
-                    }
-                    break;
-
-                case DenyAnonymousAuthorizationRequirement _:
-                    error.Append("The current user must be authenticated.");
-                    break;
-
-                case NameAuthorizationRequirement nameAuthorizationRequirement:
-                    error.Append("The current user name must match the name '");
-                    error.Append(nameAuthorizationRequirement.RequiredName);
-                    error.Append("'.");
-                    break;
-
-                case OperationAuthorizationRequirement operationAuthorizationRequirement:
-                    error.Append("Required operation '");
-                    error.Append(operationAuthorizationRequirement.Name);
-                    error.Append("' was not present.");
-                    break;
-
-                case RolesAuthorizationRequirement rolesAuthorizationRequirement:
-                    if (rolesAuthorizationRequirement.AllowedRoles == null || !rolesAuthorizationRequirement.AllowedRoles.Any())
-                    {
-                        // This should never happen.
-                        error.Append("Required roles are not present.");
-                    }
-                    else
-                    {
-                        error.Append("Required roles '");
-                        error.Append(string.Join(", ", rolesAuthorizationRequirement.AllowedRoles));
-                        error.Append("' are not present.");
-                    }
-                    break;
-
-                case AssertionRequirement _:
-                default:
-                    error.Append("Requirement '");
-                    error.Append(authorizationRequirement.GetType().Name);
-                    error.Append("' was not satisfied.");
-                    break;
-            }
+            // TODO: check to see if only a single failure result is returned in result.Failure.FailedRequirements,
+            //         and report a single specific error class for that result.  fall back to the below when multiple
+            //         errors are reported.  or instead of the fallback, report separate error instances for each entry.
```

Comment:
As already expressed in #478 I'm opposed to generating a separate error for each failed requirement. It would result in many lines starting with "You are not authorized to run this [operation] because ...", and IMO failure to authorize should be a single validation error.

Any opinions?

Reply:
I have no strong opinion. Feel free to delete the comment if that is the consensus.

Reply:
@rose-a See https://github.com/graphql-dotnet/authorization/issues/117

> It is also unnecessarily complex to query for the error code in the json, since it is an array of errors and in one query there could be errors relating to something other than authentication.

Reply:
> Any opinions?

I'm 50/50 but I'm OK to merge PR as is since `AddValidationError` is virtual now and one can override it to make multiple calls to `context.ReportError` with customized messages. Moreover then actual messages written in response could be customized in custom `IErrorInfoProvider`.

Reply:
For [graphql-dotnet/authorization#117](https://github.com/graphql-dotnet/authorization/issues/117) they would have to override [GraphQLHttpMiddleware](https://github.com/graphql-dotnet/server/blob/master/src/Transports.AspNetCore/GraphQLHttpMiddleware.cs) to create a 401 status code on an `AuthorizationError` containing a `DenyAnonymousAuthorizationRequirement` as failed requirement...

Reply:
I know. Thanks. FYI @tobias-tengler.